src/clib/pioc_sc.c

```diff
@@ -177,52 +186,60 @@ PIO_Offset GCDblocksize(int arrlen, const PIO_Offset *arr_in)
         }
     }
 
+    /* ??? */
     numblks = numtimes + 1;
     if (numtimes == 0)
         n = numblks;
     else
         n = numtimes;
 
+    /* ??? */
     bsize = (PIO_Offset)arrlen;
     if (numblks > 1)
     {
         PIO_Offset blk_len[numblks];
         PIO_Offset gaps[numtimes];
-        if(numtimes > 0)
+        if (numtimes > 0)
         {
             ii = 0;
-            for (i = 0; i < arrlen - 1; i++)
+            for (int i = 0; i < arrlen - 1; i++)
                 if (del_arr[i] > 1)
                     gaps[ii++] = del_arr[i] - 1;
             numgaps = ii;
         }
 
         j = 0;
-        for (i = 0; i < n; i++)
+        for (int i = 0; i < n; i++)
             loc_arr[i] = 1;
 
-        for (i = 0; i < arrlen - 1; i++)
+        for (int i = 0; i < arrlen - 1; i++)
             if(del_arr[i] != 1)
                 loc_arr[j++] = i;
 
         blk_len[0] = loc_arr[0];
         blklensum = blk_len[0];
-        for(i = 1; i < numblks - 1; i++)
+        for(int i = 1; i < numblks - 1; i++)
         {
             blk_len[i] = loc_arr[i] - loc_arr[i - 1];
             blklensum += blk_len[i];
         }
         blk_len[numblks - 1] = arrlen - blklensum;
 
+        /* Get the GCD in blk_len array. */
         bsize = lgcd_array(numblks, blk_len);
+
+        /* ??? */
```

Comment:
I don't recall why i needed these next two blocks, I remember struggling to get this right in all cases and I'm afraid that the end result is that bsize is almost always 1.